Xamarin.Forms.Platform.UAP/CollectionView/ItemsViewRenderer.cs

```diff
@@ -213,10 +213,11 @@ protected virtual void TearDownOldElement(ItemsView oldElement)
 				return;
 			}
 
-			if (Layout != null)
+			if (oldElement is StructuredItemsView oldStructuredItemsView)
 			{
 				// Stop tracking the old layout
-				Layout.PropertyChanged -= LayoutPropertyChanged;
+				oldStructuredItemsView.ItemsLayout.PropertyChanged -= LayoutPropertyChanged;
+				oldStructuredItemsView.ItemsLayout = null;
```

Comment:
Hey Yann! Sorry I only got to this now. I will try and think how we can validate this. I assume you tested this yourself? Are you using this code now in any of your apps by any chance?

Reply:
Hi Gerald, I use the simple project available [here](https://github.com/MADSENSE/Madsense.XamarinForms.Sample/tree/collectionview-memory-leak) with VS Diagnostics Tools to reproduce/check the fix.

For the moment, on production we use an extended renderer as workaround on many apps. Not exactly the same code, because there are private/internal on the base renderer, but the result is the same:
```cs
public class ExtendedCollectionViewRenderer : Xamarin.Forms.Platform.UWP.CollectionViewRenderer
{
    protected override void Dispose(bool disposing)
    {
        base.TearDownOldElement(Element);

        if (Element.ItemsLayout is LinearItemsLayout linearItemsLayout)
        {
            Element.ItemsLayout = new LinearItemsLayout(linearItemsLayout.Orientation);
        }
        else if (Element.ItemsLayout is GridItemsLayout gridItemsLayout)
        {
            Element.ItemsLayout = new GridItemsLayout(gridItemsLayout.Span, gridItemsLayout.Orientation);
        }
        else
        {
            Element.ItemsLayout = null;
        }

        base.Dispose(disposing);
    }

    protected override void CleanUpCollectionViewSource()
    {
        var itemContentControls = ListViewBase.GetChildren<ItemContentControl>();
        foreach (var itemContentControl in itemContentControls)
        {
            itemContentControl.FormsDataContext = null;
            itemContentControl.DataContext = null;
        }

        base.CleanUpCollectionViewSource();
    }
}
```

Reply:
When the build completes there should be NuGets produced, would you maybe be able to try them out on projects where you have this problem and see if this works as expected and doesn't break anything else? The instructions [can be found here](https://github.com/xamarin/Xamarin.Forms/wiki/Preview-Packages). That would be very helpful, thanks!

Reply:
I did some tests with the nuget produced on an application with multiple `CollectionView` on different views, the Memory Leak is solved. 
By the way, I also updated the [test project](https://github.com/MADSENSE/Madsense.XamarinForms.Sample/tree/collectionview-memory-leak).

Reply:
Thanks for the cooperation @YZahringer !